martin-mbtiles/src/tile_copier.rs

```diff
@@ -186,8 +191,39 @@ impl TileCopier {
     }
 
     async fn copy_tile_tables(&self, conn: &mut SqliteConnection) -> MbtResult<()> {
-        self.run_query_with_options(conn, "INSERT INTO tiles SELECT * FROM sourceDb.tiles")
+        if let Some(diff_file) = &self.options.diff_with_file {
+            let diff_mbtiles = Mbtiles::new(diff_file)?;
+
+            let _ = &self.detect_type(&diff_mbtiles);
+
+            query("ATTACH DATABASE ? AS newDb")
+                .bind(diff_mbtiles.filepath())
+                .execute(&mut *conn)
+                .await?;
+
+            self.run_query_with_options(
+                &mut *conn,
+                "INSERT INTO tiles
+                        SELECT COALESCE(sourceDb.tiles.zoom_level, newDb.tiles.zoom_level) as zoom_level,
+                                COALESCE(sourceDb.tiles.tile_column, newDb.tiles.tile_column) as tile_column,
+                                COALESCE(sourceDb.tiles.tile_row, newDb.tiles.tile_row) as tile_row,
+                                    newDb.tiles.tile_data as tile_data
+                        FROM sourceDb.tiles FULL OUTER JOIN newDb.tiles
+                            ON sourceDb.tiles.zoom_level = newDb.tiles.zoom_level
+                            AND sourceDb.tiles.tile_column = newDb.tiles.tile_column
+                            AND sourceDb.tiles.tile_row = newDb.tiles.tile_row
+                        WHERE (sourceDb.tiles.tile_data != newDb.tiles.tile_data
+                            OR sourceDb.tiles.tile_data ISNULL
+                            OR newDb.tiles.tile_data ISNULL)",
+            )
+            .await
+        } else {
+            self.run_query_with_options(
+                conn,
+                "INSERT INTO tiles SELECT * FROM sourceDb.tiles WHERE 1=1",
```

Comment:
please add a comment why you need `WHERE 1=1` (is it because additional clauses can be added later?).
Also, `WHERE TRUE` is cleaner - shows intent